File: Programmers/올바른 괄호.cpp
#include <string>
#include <iostream>
#include <stack>
using namespace std;

bool solution(string s)
{
    stack<char> st;
    for (int i = 0; i < s.size(); i++)
    {
        if (s[i] == '(')
            st.push(s[i]);
        else if (s[i] == ')' && !st.empty())
            st.pop();
        else
            return false;
    }
    if (st.empty())
        return true;
    else
        return false;
}

File: Programmers/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        for(int i = 0; i < s.length(); i++) {
        	if(s.charAt(i) == '(')
        		stack.add(s.charAt(i));
        	else if(s.charAt(i) == ')')
        	{
        		if(stack.empty())
        			return false;
        		else
        			stack.pop();
        	}
        }
        
        if(stack.empty())
        	return true;
        else return false;
    }
}